pinot-core/src/test/java/org/apache/pinot/core/query/optimizer/QueryOptimizerTest.java

```diff
@@ -293,6 +293,11 @@ public void testQueries() {
             + "AND TEXT_MATCH(string2, 'foo2') AND TEXT_MATCH(string2, 'bar2')",
         "SELECT * FROM testTable WHERE TEXT_MATCH(string1, '(foo1 AND bar1)') AND TEXT_MATCH(string2, '(foo2 AND "
             + "bar2)')");
+    testQuery("SELECT * FROM testTable WHERE TEXT_MATCH(string, 'foo') OR NOT TEXT_MATCH(string, 'bar')",
```

Comment:
Another option is probably optimize this to become smth like `TEXT_MATCH(string, 'foo OR (*:* not bar)')`, have we considered that?

Reply:
cc @itschrispeck

Reply:
I did consider this, but I think it often results in poorer performance. 

The generic `NOT text_match(col, 'str')` rewrite to `text_match(col, '*:* NOT str')` does solve the accuracy issues of NRT search, but in our testing significantly slowed down a large % of queries. I wasn't comfortable making the change since the optimizer is used by default. 

In general the approach of `TextMatchFilterOptimizer` only rewrites expressions that are almost always improved (the one edge case is called out in the original PR)